### ISSUE TITLE
how to use multi-thread mode?

### ISSUE THREAD

**Issue by coolit** [jgallagher/rusqlite#405]
my code is as below:

```rust
extern crate rusqlite;
extern crate time;

use time::Timespec;
use rusqlite::Connection;
use std::thread;

#[derive(Debug)]
struct Person {
    id: i32,
    name: String,
    time_created: Timespec,
    data: Option<Vec<u8>>
}


fn main() {
    let conn = Connection::open_in_memory().unwrap();

    conn.execute("CREATE TABLE person (
                  id              INTEGER PRIMARY KEY,
                  name            TEXT NOT NULL,
                  time_created    TEXT NOT NULL,
                  data            BLOB
                  )", &[]).unwrap();
    let me = Person {
        id: 0,
        name: "Steven".to_string(),
        time_created: time::get_time(),
        data: None
    };
    conn.execute("INSERT INTO person (name, time_created, data)
                  VALUES (?1, ?2, ?3)",
                 &[&me.name, &me.time_created, &me.data]).unwrap();
    conn.execute("INSERT INTO person (name, time_created, data)
                  VALUES (?1, ?2, ?3)",
                 &[&"Tom".to_string(), &me.time_created, &me.data]).unwrap();


    let mut thread_holder = vec![];

    for i in 0..2 {
        let conn_ref = &conn;
        thread_holder.push(thread::spawn(move || {
            let keyword = "SELECT id, name, time_created, data FROM person";
            let mut stmt = conn_ref.prepare(keyword).unwrap();
            let mut person_iter = stmt.query_map(&[], |row| {
                Person {
                    id: row.get(0),
                    name: row.get(1),
                    time_created: row.get(2),
                    data: row.get(3)
                }
            }).unwrap();
            let a = person_iter.next();
            a;
        }));
    }
    for thread_elements in thread_holder {
        println!("Thread returned {:?}",
                 thread_elements.join().unwrap());
    }


}
```

**Comment by gwenn** [comment]
You have to open one connection by thread.
See https://github.com/jgallagher/rusqlite/issues/188

**Comment by coolit** [comment]
So multi-thread is not yet supported in rusqlite?

**Comment by gwenn** [comment]
I guess you didn't read #188:
http://sqlite.org/threadsafe.html:
> _Multi-thread_. In this mode, SQLite can be safely used by multiple threads provided that **no single database connection is used simultaneously in two or more threads**.
So `rusqlite` supports SQLite multi-thread mode by preventing two threads to reference the same connection simultaneously.

**Comment by coolit** [comment]
I see. I am still new with Rust. Hopefully there can be more examples available for newcomers to learn.

**Comment by gwenn** [comment]
There is a pending issue that may help you when fixed: #393.
You will be able to replace:
```rust
let conn_ref = &conn;
```
by
```rust
let conn_ref = conn.clone(); // or a better name
```
to make your program works.
But currently, there is no such solution.

**Comment by coolit** [comment]
Thanks. Looking forward to the clone feature.
